site/docs/glossary.md

```diff
@@ -0,0 +1,537 @@
+so the dynamic execution strategy provides the best possible incremental and
+full build times.
+
+#### Execution phase
+
+The third phase of a build. Executes the [*actions*](#action) from the action
+graph created during the analysis phase. Executed actions invoke executables
+(compilers, scripts) to read and write [*artifacts*](#artifact). *Spawn
+strategies* control how these actions are executed: locally, remotely,
+dynamically, sandboxed, docker, and so on.
+
+#### Execution root
+
+A directory in the workspace’s output base directory where local actions are
+executed in non-sandboxed builds. The directory contents are mostly symlinks of
+input [*artifacts*](#artifact) from the workspace. The execution root also
+contains symlinks to external repositories as other inputs and the `bazel-out`
+directory to store outputs. Prepared during the loading phase by planting a
+"symlink forest" of the directories representing transitive closure of packages
+depended on by the target(s) to build. Accessible with `bazel info
+execution_root` on the command line.
+
+#### File
+
+See [*Artifact*](#artifact).
+
+#### Hermeticity
+
+The property of a build or test where there are no external influences on the
+build or test. This helps to enable deterministic and correct results. This
+usually means disallowing network access to actions, only allowing access to
+declared input [*artifacts*](#artifact), making timestamps and timezones fixed,
+restricting access/resetting environment variables, and using fixed seeds for
+random number generators.
+
+#### Incremental build
+
+A build that reuses the results of earlier builds to reduce build time and
+resource needs. Bazel aims to guarantee correct results for incremental builds.
+This is the opposite of a clean build.
+
+<!-- TODO: #### Install base -->
+
+#### Label
+
+An identifier for a target. A fully qualified label such as
+`//path/to/package:target` consists of `//` to mark the workspace root
+directory, path/to/package as the directory that contains the [*BUILD
+file*](#build-file) declaring the target, and :target as the name of the target
+declared in the aforementioned BUILD file. May also be prefixed with
+`@my_repository//<..>` to indicate that the target is declared in an *external
+repository* named `my_repository`.
+
+#### Loading phase
+
+The first phase of a build. WORKSPACE, BUILD, and .bzl files are parsed to
+create *packages*. Macros and certain functions like glob() are evaluated in
+this phase. Interleaved with the second phase of the build, the *analysis
+phase*, to build up a *target dependency graph*.
+
+#### Macro
+
+A mechanism to compose multiple rule target declarations together under a single
+Starlark function. Enables reusing common rule declaration patterns across BUILD
+files. Expanded to the underlying rule target declarations during the *loading
+phase*.
+
+#### Mnemonic
+
+A short, human-readable string to quickly understand what an *action* is doing.
```

Comment:
A couple of notes here:

The context of the term is not clear -- where is the mnemonic declared? In BUILD files, in .bzl files? Second, it's not clear who is supposed to be understanding when worded like this.

For example, this might better be something like:

"A *mnemonic* is a short, human-readable string in a `.bzl` file that describes what an action is doing. Mnemonics can be used as identifiers ..."

Reply:
Thanks! I've reworded it to make it clearer. PTAL.